contracts/Governance/TimelockV8.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+/**
+ * @title TimelockV8
+ * @author Venus
+ * @notice The Timelock contract using solidity V8.
+ * This contract also differs from the original timelock because it has a virtual function to get minimum delays
+ * and allow test deployments to override the value.
+ */
+contract TimelockV8 {
+    /// @notice Event emitted when a new admin is accepted
+    event NewAdmin(address indexed newAdmin);
+
+    /// @notice Event emitted when a new admin is proposed
+    event NewPendingAdmin(address indexed newPendingAdmin);
+
+    /// @notice Event emitted when a new admin is proposed
+    event NewDelay(uint256 indexed newDelay);
+
+    /// @notice Event emitted when a proposal transaction has been cancelled
+    event CancelTransaction(
+        bytes32 indexed txHash,
+        address indexed target,
+        uint256 value,
+        string signature,
+        bytes data,
+        uint256 eta
+    );
+
+    /// @notice Event emitted when a proposal transaction has been executed
+    event ExecuteTransaction(
+        bytes32 indexed txHash,
+        address indexed target,
+        uint256 value,
+        string signature,
+        bytes data,
+        uint256 eta
+    );
+
+    /// @notice Event emitted when a proposal transaction has been queued
+    event QueueTransaction(
+        bytes32 indexed txHash,
+        address indexed target,
+        uint256 value,
+        string signature,
+        bytes data,
+        uint256 eta
+    );
+
+    /// @notice Required period to execute a proposal transaction
+    uint256 private constant GRACE_PERIOD = 14 days;
+
+    /// @notice Minimum amount of time a proposal transaction must be queued
+    uint256 private constant MINIMUM_DELAY = 1 hours;
+
+    /// @notice Maximum amount of time a proposal transaction must be queued
+    uint256 private constant MAXIMUM_DELAY = 30 days;
+
+    /// @notice Timelock admin authorized to queue and execute transactions
+    address public admin;
+
+    /// @notice Account proposed as the next admin
+    address public pendingAdmin;
+
+    /// @notice Period for a proposal transaction to be queued
+    uint256 public delay;
+
+    /// @notice Mapping of queued transactions
+    mapping(bytes32 => bool) public queuedTransactions;
+
+    constructor(address admin_, uint256 delay_) public {
```

Comment:
Private constructors are deprecated now anyway in favor of `abstract contract`, so no need to specify visibility

```suggestion
    constructor(address admin_, uint256 delay_) {
```